.github/workflows/cache-build.yaml

```diff
@@ -0,0 +1,141 @@
+      - name: Run integration tests on amd64 to update docker cache
+        uses: docker/build-push-action@v2
+        id: docker_tests_ci_cache_update
+        with:
+          context: .
+          file: ./Dockerfile.tests
+          platforms: linux/amd64
+          build-args: |
+            BUILDER_IMAGE=quay.io/${{ github.repository_owner }}/cilium-envoy-builder:test-archive-latest
+            BAZEL_BUILD_OPTS=--remote_upload_local_results=false
+            TESTLOGS=bazel-out/k8-fastbuild/testlogs
+          cache-to: type=local,dest=/tmp/buildx-cache,mode=max
+          push: false
+  build-cache-and-push-images:
+    timeout-minutes: 360
+    name: Build cache and push images
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Add swap
+        run: |
+          set -x
+          file="/var/lib/swap"
+          sudo fallocate -l 4G "${file}"
+          sudo chmod 600 "${file}"
+          sudo mkswap "${file}"
+          sudo swapon "${file}"
+          sudo swapon -s
+          free -m
+          df -h
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_ENVOY_USERNAME }}
+          password: ${{ secrets.QUAY_ENVOY_PASSWORD }}
+      - name: Checkout PR
+        uses: actions/checkout@v2
+      - name: Prep for build
+        run: echo "${{ github.sha }}" >SOURCE_VERSION
+      - name: Multi-arch build & push of build artifact archive
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          target: builder-archive
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            COPY_CACHE_EXT=.new
+            BUILDER_IMAGE=quay.io/${{ github.repository_owner }}/cilium-envoy-builder:envoy-1.17.0-archive
+          push: true
+          tags: quay.io/${{ github.repository_owner }}/cilium-envoy-builder:release-archive-latest
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/buildx-cache
+          key: docker-cache-master
+      - name: Clear cache
+        run: |
+          rm -rf /tmp/buildx-cache/*
+          docker buildx prune -f
+      - name: Multi-arch build & push master latest
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            BAZEL_BUILD_OPTS=--remote_upload_local_results=false
+            BUILDER_IMAGE=quay.io/${{ github.repository_owner }}/cilium-envoy-builder:release-archive-latest
+          cache-to: type=local,dest=/tmp/buildx-cache,mode=max
+          push: true
+          tags: |
+            quay.io/${{ github.repository_owner }}/cilium-envoy:latest
+            quay.io/${{ github.repository_owner }}/cilium-envoy:${{ github.sha }}
```

Comment:
same for the sha. (don't forget to add an ID to this step)

Reply:
Same.